dags/backfill.py

```diff
@@ -0,0 +1,130 @@
+import ast
+import datetime
+from typing import Optional
+
+from airflow.decorators import dag
+from airflow.operators.bash import BashOperator
+from airflow.operators.dummy import DummyOperator
+from airflow.operators.python import PythonOperator, BranchPythonOperator
+from airflow.utils.trigger_rule import TriggerRule
+from airflow.models import DagModel
+
+
+from utils.backfill import BackfillParams
+from utils.tags import Tag
+
+def __parse_string_params(string_params: str) -> Optional[BackfillParams]:
+    """
+    dag_run.conf is string representation of a Python dictionary in Airflow 2.1
+    ast.literal_eval() is used to convert from string to dictionary as a workaround
+    this workaround will no longer be required in Airflow >=2.2, see link below for future implementation
+    https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html
+    """
+    params_parsed = ast.literal_eval(string_params)
+    return BackfillParams(**params_parsed)
+
+
+def dry_run_branch_callable(params: str) -> str:
+    backfill_params = __parse_string_params(params)
+    return "dry_run" if backfill_params.dry_run else "real_deal"
+
+
+def clear_branch_callable(params: str) -> str:
+    backfill_params = __parse_string_params(params)
+    return "clear_tasks" if backfill_params.clear else "do_not_clear_tasks"
```

Comment:
same comment here, we want to have vars here to compare instead of hardcoded strings